readthedocs/projects/tasks.py

```diff
@@ -127,6 +128,10 @@ def run(self, pk, version_pk=None, build_pk=None, record=True, docker=False,
 
             self.config = load_yaml_config(version=self.version)
 
+        if self.setup_env.failed:
+            self.send_notifications()
+            raise BuildEnvironmentError(_('Build Setup Failed'))
```

Comment:
I think returning is enough? There isn't anything to catch the `BuildEnvironmentError` here, and setup_env should set the build state correctly.